diff --git a/sys/_types.h b/sys/_types.h
index fe345b43d5eb..8296cbdf2e1e 100644
--- a/sys/_types.h
+++ b/sys/_types.h
@@ -179,11 +179,9 @@ typedef	__uint_least32_t __char32_t;
 #endif
 
 typedef struct {
-	long long __max_align1
-	    __attribute__((__aligned__(__alignof__(long long))));
+	long long __max_align1;
 #ifndef _STANDALONE
-	long double __max_align2
-	    __attribute__((__aligned__(__alignof__(long long))));
+	long double __max_align2;
 #endif
 } __max_align_t;
 
