diff --git endian.h endian.h
index 3b077cd9295..cb6948d3fec 100644
--- endian.h
+++ endian.h
@@ -31,6 +31,8 @@
 #include <sys/_types.h>
 #endif
 
+#ifdef __GNUC__
+
 static __inline __uint16_t
 __swap16md(__uint16_t _x)
 {
@@ -55,6 +57,8 @@ __swap64md(__uint64_t _x)
 /* Tell sys/endian.h we have MD variants of the swap macros.  */
 #define __HAVE_MD_SWAP
 
+#endif /* __GNUC__ */
+
 #define _BYTE_ORDER _LITTLE_ENDIAN
 
 #ifndef __FROM_SYS__ENDIAN
