--- machine/endian.h.orig	Sat Sep  5 19:21:50 2020
+++ machine/endian.h	Sat Sep  5 19:22:51 2020
@@ -31,6 +31,8 @@
 #include <sys/_types.h>
 #endif
 
+#ifdef __GNUC__
+
 static __inline __uint16_t
 __swap16md(__uint16_t _x)
 {
@@ -54,6 +56,8 @@
 
 /* Tell sys/endian.h we have MD variants of the swap macros.  */
 #define __HAVE_MD_SWAP
+
+#endif
 
 #define _BYTE_ORDER _LITTLE_ENDIAN
 
--- machine/_types.h.orig	Sat Mar 23 19:04:19 2024
+++ machine/_types.h	Sat Mar 23 19:04:36 2024
@@ -122,11 +122,7 @@
 typedef long			__ptrdiff_t;
 typedef	unsigned long		__size_t;
 typedef	long			__ssize_t;
-#if defined(__GNUC__) && __GNUC__ >= 3
 typedef	__builtin_va_list	__va_list;
-#else
-typedef char *			__va_list;
-#endif
 
 /* Wide character support types */
 #ifndef __cplusplus
